### PR TITLE
feat: add errors to user settings

### PR DESCRIPTION
**Pull request by benfurber**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Building off #2717, this adds the errors component to the user settings form

## Git Issues

Part of #2684

### CODE REVIEW

File: src/pages/Settings/content/formSections/SettingsErrors.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { ErrorsContainer } from 'src/common/Form/ErrorsContainer'
+import { defaultError, fields } from 'src/pages/Settings/labels'
+
+import type { ValidationErrors } from 'final-form'
+
+const flattenErrors = (formErrors) => {
+  const errors = {}
+
+  Object.keys(formErrors).forEach((key) => {
+    if (typeof formErrors[key] !== 'string') return (errors[key] = defaultError)
```
Comment (evakill) [inline src/pages/Settings/content/formSections/SettingsErrors.tsx:10]: why do we have to do this check for the settings?

Comment (benfurber) [reply src/pages/Settings/content/formSections/SettingsErrors.tsx:10]: [It relates to this comment](https://github.com/ONEARMY/community-platform/issues/2684#issuecomment-1688150598). There's a couple of weirdly shapped errors:

There's `coverImages` which includes `FINAL_FORM/array-error: "Must have at least one cover image"`

And I think it's ok to flatten the two errors for an individual link into one.
```
        "links": [
            {
                "label": "Make sure this field is filled correctly",
                "url": "Make sure this field is filled correctly"
            }
```
into: Contacts & links: Make sure this field is filled correctly.

**Review by evakill — APPROVED** [review]
sorry for the delay! the code looks good, please rebase and then we can merge.

**Comment by benfurber** [comment]
@evakill Done.

**Pull request merged**